test/e2e/nodepool_set_test.go

```diff
@@ -0,0 +1,42 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+
+	. "github.com/onsi/gomega"
+)
+
+var (
+	zeroReplicas int32 = 0
+	oneReplicas  int32 = 1
+	twoReplicas  int32 = 2
+)
+
+func TestNodePoolMain(t *testing.T) {
```

Comment:
let's call this just testNodePool so it's clear it's just another test, main can be confused with our TestMain which ultimately calls the built-in m *testing.M